costmap_2d/include/costmap_2d/layer.h

```diff
@@ -116,6 +116,18 @@ class Layer
     return false;
   }
 
+  virtual bool isStaticLayer() {
+    return false;
+  }
+
+  virtual std::shared_ptr<std::vector<double>> getDistancesFromStaticMap() {
+    return std::shared_ptr<std::vector<double>>();
```

Comment:
Should create a static empty std::shared_ptr<std::vector<double>> for efficiency.